frontend/lib/src/components/widgets/DataFrame/hooks/useSelectionHandler.ts

```diff
@@ -157,9 +160,23 @@ function useSelectionHandler(
           ...updatedSelection,
           rows: gridSelection.rows,
         }
+
         syncSelection = true
       }
 
+      if (columnSelectionChanged && updatedSelection.columns.length >= 0) {
+        columns.forEach((column, idx) => {
+          if (column.isIndex) {
+            // Remove all index columns from the column selection
+            // We don't want to allow selection of index columns.
+            updatedSelection = {
+              ...updatedSelection,
+              columns: updatedSelection.columns.remove(idx),
```

Comment:
we should probably here just remove the columns and then build the object outside of the loop, because I guess `{...updatedSelection, columns...}` is a rather expensive operation

Reply:
Yeah, but 99% of dataframes have 1 index column so it would not make a difference

Reply:
And this is not a performance critical operation since it's not part of the rendering

Reply:
I added a small improvement